dashboard/modules/job/job_head.py

```diff
@@ -103,6 +107,23 @@ async def upload_package(self, req: Request):
 
         return Response(status=aiohttp.web.HTTPOk.status_code)
 
+    @routes.post("/api/packages/{protocol}/{package_name}/ref")
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=False)
+    async def add_temporary_uri_reference(self, req: Request):
+        package_uri = http_uri_components_to_uri(
+            protocol=req.match_info["protocol"],
+            package_name=req.match_info["package_name"],
+        )
+        logger.debug(f"Adding temporary reference to package {package_uri}.")
+        try:
+            add_temporary_uri_reference(package_uri)
+        except Exception:
+            return Response(
+                text=traceback.format_exc(),
+                status=aiohttp.web.HTTPInternalServerError.status_code,
+            )
+        return Response(status=aiohttp.web.HTTPOk.status_code)
```

Comment:
I don't think we need to add a separate API endpoint for this (we should ideally treat API endpoints are ~~ public APIs), I'd prefer if we just do this in the existing API endpoint we use to check if the ref exists.

Reply:
Good point, that's cleaner.